src/coap_client.c

```diff
@@ -26,6 +26,46 @@ bool golioth_client_is_connected(struct golioth_client *client)
     return client->session_connected;
 }
 
+void golioth_coap_next_token(uint8_t token[GOLIOTH_COAP_TOKEN_LEN])
+{
+    static golioth_sys_mutex_t token_mut;
+    static uint8_t stored_token[GOLIOTH_COAP_TOKEN_LEN] = {0};
+
+    if (!token_mut)
+    {
+        token_mut = golioth_sys_mutex_create();
+
+        uint32_t rnd = 0;
+        golioth_sys_srand(golioth_sys_now_ms());
```

Comment:
This relates to my previous reply that seeding "lazily" inserts a variable amount of time due to awaiting the network connection to come up before generating the seed. In my testing this works well except for native_sim (and probably Linux but I don't remember) which don't need to wait for the network and always have the same first seed.

Reply:
Got it, that's pretty clever. Can we (or have you) test it on the RT1024? I wonder if a hardwired connection has less variability in connection time.

Reply:
I tested this again with the 1024 and I wasn't able to get two of the same starting tokens in a row, but trying multiple times I do see repeats. I guess the variability is probably within a handful of milliseconds so not much randomness. As you mentioned, this will be resolved when we connect golioth_sys_rand() to the Zephyr random function instead of the C standard rand().